src/tools/rbd/action/Import.cc

```diff
@@ -132,6 +138,30 @@ static int do_import(librbd::RBD &rbd, librados::IoCtx& io_ctx,
 #endif
   }
 
+  if (import_snaps || !from_stdin) {
+    r = safe_read_exact(fd, image_buf, utils::RBD_IMAGE_BANNER.size());
+    if (r < 0)
+      goto done;
+    image_buf[utils::RBD_IMAGE_BANNER.size()] = '\0';
+    if (strcmp(image_buf, utils::RBD_IMAGE_BANNER.c_str())) {
+      // Old format
```

Comment:
okey, similar with above comment, agree with you. We can make it working in a much simpler way.